annotto-front/src/modules/project/components/common/image/ImageContainer.js

```diff
@@ -246,37 +251,36 @@ const ImageContainer = ({
       >
         <Layer draggable={!selectedSection}>
           <Image image={image} ref={_onImageRefChange} />
-          {imageHeight
-            ? resolvedAnnotationsAndPredictions.map(({ zone, value, annotationIndex, predictionIndex }, index) => {
-                const task = tasks.find((t) => t.value === value)
-                if (task && !task?.color) {
-                  task.color = theme.colors.defaultAnnotationColors[tasks.indexOf(task)]
-                }
-
-                return (
-                  <ZoneMarker
-                    key={index}
-                    index={index}
-                    task={task}
-                    imageHeight={imageHeight}
-                    imageWidth={imageWidth}
-                    annotationIndex={annotationIndex}
-                    predictionIndex={predictionIndex}
-                    zone={zone}
-                    value={value}
-                    scale={stageRef.current?.scaleX()}
-                    isSelected={selectRoomId === index}
-                    onDeleteClick={_onDeleteClick(
-                      isNumber(predictionIndex) && !annotationIndex ? predictionIndex : annotationIndex
-                    )}
-                    onValidateClick={_onValidateClick(predictionIndex)}
-                    onDragEnd={_onDragImageEnd(zone, value)}
-                    onTransformEnd={_onTransformEnd(zone, value)}
-                    onSelectClick={_onSelectRoomId(index)}
-                  />
-                )
-              })
-            : null}
+          {!!imageHeight &&
+            resolvedAnnotationsAndPredictions.map(({ zone, value, annotationIndex, predictionIndex }, index) => {
```

Comment:
I think its not a good practice to have JS const declaration within your JSX. You should create a new component instead, which will make your code easier to read and maintain too !